src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -75,6 +78,30 @@ object TaskGroupBuilder {
     Some((executorInfo.build, taskGroup.build, resourceMatch.hostPorts))
   }
 
+  // The resource match provides us with a list of host ports.
+  // Each port mapping corresponds to an item in that list.
+  // We use that list to swap the dynamic ports (ports set to 0) with the matched ones.
+  private[this] def computePortMappings(
+    endpoints: Seq[raml.Endpoint],
+    hostPorts: Seq[Option[Int]]): Seq[mesos.NetworkInfo.PortMapping] = {
+
+    endpoints.zip(hostPorts).collect {
+      case (mapping, Some(hostPort)) =>
+        if (mapping.containerPort.isEmpty || mapping.containerPort.get == 0) {
+          mesos.NetworkInfo.PortMapping.newBuilder
+            .setHostPort(hostPort)
+            .setContainerPort(hostPort).build
+        } else {
+          val portMapping = mesos.NetworkInfo.PortMapping.newBuilder
+            .setHostPort(hostPort)
+          mapping.containerPort.foreach(portMapping.setContainerPort)
+
+          // TODO(nfnt): set the protocol
```

Comment:
.. and there might in multiple protocols, which case the mapping is (nearly) duplicated for each protocol.